### PR TITLE
Add approvaltests.cpp 8.1.1 and 8.2.0

### PR DESCRIPTION
**Pull request by claremacrae**
Specify library name and version:  **approvaltests.cpp/8.1.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Tested in my local fork with this - after installing hooks:

```bash
CONAN_HOOK_ERROR_LEVEL=40 conan create . 8.1.1@
```

(This is a header-only library)

### CODE REVIEW

**Comment by uilianries** [comment]
Hi @claremacrae ! Thanks for updating it!

Could you please comment your early access program request on #4 ? The PR bot only allows users which are approved (manual action) on #4.

Best regards!

**Comment by claremacrae** [comment]
> Could you please comment your early access program request on #4 ? The PR bot only allows users which are approved (manual action) on #4.

Thanks - done

**Comment by claremacrae** [comment]
I've rebased to pull in changes made since I updated my fork. Code is unchanged.

**Comment by danimtb** [comment]
@claremacrae please add the new version to the _config.yml_ file as well

**Comment by claremacrae** [comment]
@danimtb OK - Working on it...

**Comment by claremacrae** [comment]
> @claremacrae please add the new version to the _config.yml_ file as well

@danimtb done... (it took a little while to automate the creation of the new text)

**Comment by danimtb** [comment]
@claremacrae you have to update this file as well https://github.com/conan-io/conan-center-index/blob/master/recipes/approvaltests.cpp/config.yml so packages for the new versions are generated in this PR

File: recipes/approvaltests.cpp/config.yml [hunk 1]
```diff
@@ -3,3 +3,7 @@ versions:
     folder: all
   8.1.0:
     folder: all
+  8.1.2:
```
Comment (uilianries) [inline recipes/approvaltests.cpp/config.yml]:
```suggestion
  8.1.1:
```

Comment (claremacrae) [reply recipes/approvaltests.cpp/config.yml]: Thanks @uilianries - I've pushed the edit...
Am currently working on scripting the editing of this file for future releases

**Comment by claremacrae** [comment]
> _config.yml_ syntax error in build 6:
> 
> ```
> Wrong YAML format processing '8.0.0' key|
> ```

I don't understand what this error is, and can't see the log

**Comment by danimtb** [comment]
taking a look, thanks

**Comment by danimtb** [comment]
Fix has been deployed. Relaunching the CI for this PR!

**Comment by claremacrae** [comment]
> Fix has been deployed. Relaunching the CI for this PR!

Thanks @danimtb !

**Comment by claremacrae** [comment]
Thanks for the help everyone - For future releases, the creation of pull requests will be pretty much automated, so should be painless!

**Pull request merged**